react/AddToCartButton.tsx

```diff
@@ -121,15 +130,26 @@ function AddToCartButton(props: Props) {
 
     const action = success
       ? {
-          label: translateMessage(messages.seeCart),
-          href: customToastUrl,
-        }
+        label: translateMessage(messages.seeCart),
+        href: customToastUrl,
+      }
       : undefined
 
     showToast({ message, action })
   }
 
   const handleAddToCart: React.MouseEventHandler = event => {
+
+    if (onClickBehavior === "go-to-product-page" && productLink.linkText && productLink.productId) {
+      return navigate({
+        page: 'store.product',
+        params: {
+          slug: productLink.linkText,
+          id: productLink.productId,
+        }
+      })
+    }
```

Comment:
You have to stop the event before call `navigate` too
```suggestion
    if (onClickBehavior === "go-to-product-page" && productLink.linkText && productLink.productId) {
      event.stopPropagation()
      event.preventDefault()
      return navigate({
        page: 'store.product',
        params: {
          slug: productLink.linkText,
          id: productLink.productId,
        }
      })
    }
```